Frameworks/UIKit/UIImage.mm

```diff
@@ -889,22 +892,34 @@ + (UIImage*)animatedResizableImageNamed:(NSString*)name capInsets:(UIEdgeInsets)
     return StubReturn();
 }
 
+static NSData* _ObtainImageRepresentation(UIImage* img, CFStringRef type, NSDictionary* properties) {
+    RETURN_NULL_IF(!img);
+    CFMutableDataRef cfData = CFDataCreateMutable(nullptr, 0);
+    CFAutorelease(cfData);
+    CGImageDestinationRef destination = CGImageDestinationCreateWithData(cfData, type, 1, nullptr);
+    CFAutorelease(destination);
+    CGImageDestinationAddImage(destination, getImage(img), static_cast<CFDictionaryRef>(properties));
+    if (!CGImageDestinationFinalize(destination)) {
+        return nil;
+    }
+    return static_cast<NSData*>(cfData);
+}
+
 /**
- @Status Stub
+ @Status Interoperable
 */
 NSData* UIImagePNGRepresentation(UIImage* img) {
```

Comment:
I have test for it but looks like our PNG representation ends up being a bit off vs the actual PNG data on disk (I took it out for now, i'll update the CR)

Note this function is what is used in WOCCataglog paste and it works fine.